clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6967,6 +6967,11 @@ static void checkAttributesAfterMerging(Sema &S, NamedDecl &ND) {
         } else if (isa<CXXConstructorDecl>(MD) || isa<CXXDestructorDecl>(MD)) {
           S.Diag(A->getLocation(), diag::err_lifetimebound_ctor_dtor)
               << isa<CXXDestructorDecl>(MD) << A->getRange();
+        } else if (FD->getReturnType()->isVoidType()) {
```

Comment:
nit: use `MD` instead of `FD` since we are working on a method decl (although both are correct).

Reply:
Done.